File: recipe-service/src/main/java/com/demeter/recipeservice/dto/RecipeResponse.java
package com.demeter.recipeservice.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RecipeResponse {
    private Long id;
    private String name;
    private Integer servingSize;
    private String author;
    private String authorSub;
    private BigDecimal price;
    private List<IngredientResponse> ingredientList;
    private List<StepResponse> method;
    private Integer preparationTime;
    private Integer cookingTime;
    private Integer restingTime;
    private Integer readyToServe;
    private PhotoResponse photo;
    private Integer likes;
    private Integer disLikes;
    private List<CommentResponse> commentResponseList;
    private boolean sketch;
}

File: demeter-ui/src/app/home/recipeResponse.ts
import {Ingredient} from "../add-recipe/ingredient";
import {Step} from "../add-recipe/step";
import {UploadPhotoResponse} from "../add-recipe/uploadPhotoResponse";
import {CommentRequest} from "../comments/commentRequest";

export class RecipeResponse{
  constructor(
    public id: number,
    public name: string,
    public author: string,
    public authorSub: string,
    public servingSize: number,
    public ingredientList: Ingredient[],
    public method: Step[],
    public price: number,
    public preparationTime: number,
    public cookingTime: number,
    public restingTime: number,
    public photo: UploadPhotoResponse,
    public likes: number,
    public disLikes: number,
    public comments: CommentRequest[],
    public sketch: boolean
  ){}

  markAsRecipe() {
    this.sketch = false;
    return this;
  }
}
